integration-tests/tests/circuits.rs

```diff
@@ -7,64 +7,91 @@ use integration_tests::log_init;
 use paste::paste;
 
 macro_rules! declare_tests {
-    ($name:ident, $block_tag:expr) => {
+    ($name:ident, $block_tag:expr, $mock:expr) => {
         paste! {
             #[tokio::test]
             async fn [<serial_test_evm_ $name>]() {
                 log_init();
                 let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
-                test_evm_circuit_block(*block_num, true).await;
+                test_evm_circuit_block(*block_num, $mock).await;
             }
 
             #[tokio::test]
             async fn [<serial_test_state_ $name>]() {
                 log_init();
                 let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
-                test_state_circuit_block(*block_num, true).await;
+                test_state_circuit_block(*block_num, $mock).await;
             }
 
             #[tokio::test]
             async fn [<serial_test_tx_ $name>]() {
                 log_init();
                 let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
-                test_tx_circuit_block(*block_num, true).await;
+                test_tx_circuit_block(*block_num, $mock).await;
             }
 
             #[tokio::test]
             async fn [<serial_test_bytecode_ $name>]() {
                 log_init();
                 let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
-                test_bytecode_circuit_block(*block_num, true).await;
+                test_bytecode_circuit_block(*block_num, $mock).await;
             }
 
             #[tokio::test]
             async fn [<serial_test_copy_ $name>]() {
                 log_init();
                 let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
-                test_copy_circuit_block(*block_num, true).await;
+                test_copy_circuit_block(*block_num, $mock).await;
             }
         }
     };
 }
 
-declare_tests!(circuit_block_transfer_0, "Transfer 0");
+// Mock tests
+declare_tests!(circuit_block_transfer_0_mock, "Transfer 0", true);
```

Comment:
I see that now the test functions names look like `foo` and `foo_mock` for each case. This will not be very friendly towards selecting only one set of tests.  Before this was easy because we had two features to select one set or the other: `circuits_actual` and `circuits_mock`, but also the test functions were in different modules (where as now they are all in the same module).

So I would suggest to keep the test functions in two separate modules so it's easy to test each set separately.  Also, currently there's 2 macro calls for each test block: I think this can be improved!

So here are my suggestions:
1. Remove the `#![cfg(feature = "circuits_actual")]` at the top because I think now it doesn't make sense.
2. Group the tests into two modules: `mock_prover` and `real_prover`
3. Add another macro so that we don't need to write twice the code for each block.  I think this could be achieved with 1 new macro that takes a list of `$name, $block_tag` and unrolls them two times: one inside `mod mock_prover {}` and the ohter in `mod real_prover {}`.  And in each module, have calls to `declare_tests!` each input in the list (in the first mod with true, the other one with false).  The unrolling can follow this example https://users.rust-lang.org/t/unroll-macro-argument-list/10510/2